sql/core/src/main/scala/org/apache/spark/sql/SparkSessionExtensions.scala

```diff
@@ -136,13 +146,21 @@ class SparkSessionExtensions {
   }
 
   /**
-   * Inject a rule that can override the the query stage preparation phase of adaptive query
+   * Inject a rule that can override the query stage preparation phase of adaptive query
    * execution.
    */
-  def injectQueryStagePrepRule(builder: QueryStagePrepRuleBuilder): Unit = {
+  def injectQueryStagePrepRule(builder: PhysicalPlanRuleBuilder): Unit = {
```

Comment:
`injectQueryStagePrepRule ` becomes inconsistent now from the builder, `PhysicalPlanRuleBuilder`.

Reply:
Ditto. Let's preserve this.

Reply:
I was looking to make the name of this more generic, but I can see that this causes unnecessary change so will revert this.

Reply:
Yep. Thank you. We want to minimize the breaking API change as much as possible for RAPID users.